java/runtime/src/main/java/io/ray/runtime/runtimeenv/RuntimeEnvImpl.java

```diff
@@ -1,63 +1,93 @@
 
-  private Map<String, String> envVars = new HashMap<>();
+  private static ObjectMapper mapper = new ObjectMapper();
 
-  private List<String> jars = new ArrayList<>();
+  public ObjectNode runtimeEnvs = mapper.createObjectNode();
 
-  public RuntimeEnvImpl(Map<String, String> envVars, List<String> jars) {
-    this.envVars = envVars;
-    if (jars != null) {
-      this.jars = jars;
-    }
-  }
+  public RuntimeEnvImpl() {}
 
-  public Map<String, String> getEnvVars() {
-    return envVars;
+  @Override
+  public void set(String name, Object value) {
+    JsonNode node = mapper.valueToTree(value);
+    runtimeEnvs.set(name, node);
   }
 
-  public List<String> getJars() {
-    return jars;
+  @Override
+  public void setJsonStr(String name, String jsonStr) throws RuntimeEnvException {
+    JsonNode node = null;
+    try {
+      node = JsonLoader.fromString(jsonStr);
+    } catch (IOException e) {
+      throw new RuntimeException(e);
+    }
+    runtimeEnvs.set(name, node);
   }
 
   @Override
-  public String toJsonBytes() {
-    // Get serializedRuntimeEnv
-    String serializedRuntimeEnv = "{}";
-
-    Map<String, Object> runtimeEnvMap = new HashMap<>();
-    if (!envVars.isEmpty()) {
-      runtimeEnvMap.put("env_vars", envVars);
+  public <T> T get(String name, Class<T> classOfT) throws RuntimeEnvException {
+    try {
+      return mapper.treeToValue(runtimeEnvs.get(name), classOfT);
+    } catch (JsonProcessingException e) {
+      throw new RuntimeException(e);
     }
-    if (!jars.isEmpty()) {
-      runtimeEnvMap.put("java_jars", jars);
+  }
+
+  @Override
+  public String getJsonStr(String name) throws RuntimeEnvException {
+    try {
+      return mapper.writeValueAsString(runtimeEnvs.get(name));
+    } catch (JsonProcessingException e) {
+      throw new RuntimeException(e);
     }
+  }
 
-    serializedRuntimeEnv = new Gson().toJson(runtimeEnvMap);
+  @Override
+  public void remove(String name) {
+    runtimeEnvs.remove(name);
+  }
 
-    // Get serializedRuntimeEnvInfo
-    if (serializedRuntimeEnv.equals("{}") || serializedRuntimeEnv.isEmpty()) {
-      return "{}";
-    }
+  @Override
+  public String serialize() throws RuntimeEnvException {
+    // TODO(SongGuyang): Expose runtime env config API to users.
     RuntimeEnvCommon.RuntimeEnvInfo.Builder protoRuntimeEnvInfoBuilder =
         RuntimeEnvCommon.RuntimeEnvInfo.newBuilder();
-    protoRuntimeEnvInfoBuilder.setSerializedRuntimeEnv(serializedRuntimeEnv);
+    try {
+      protoRuntimeEnvInfoBuilder.setSerializedRuntimeEnv(mapper.writeValueAsString(runtimeEnvs));
+    } catch (JsonProcessingException e) {
+      throw new RuntimeException(e);
+    }
     JsonFormat.Printer printer = JsonFormat.printer();
     try {
       return printer.print(protoRuntimeEnvInfoBuilder);
     } catch (InvalidProtocolBufferException e) {
       throw new RuntimeException(e);
     }
   }
+
+  public RuntimeEnvCommon.RuntimeEnvInfo GenerateRuntimeEnvInfo() throws RuntimeEnvException {
```

Comment:
There is no any usage for `GenerateRuntimeEnvInfo`.

Reply:
Here https://github.com/ray-project/ray/pull/26783/files#diff-de3a842e353216ddd6d48f5eb9da067f4c3936a7444f1df64945f6a4cc1baf25R111 ?